product-service/src/functions/products-by-id/handler.test.ts

```diff
@@ -28,24 +28,24 @@ describe('getProductsByIdHandler', () => {
       "title": "Coco",
       "posterPath": "/gGEsBPAijhVUFoiNpgZXqRVWJt2.jpg"
     };
-    jest.spyOn(productsActions, 'getProductsById').mockImplementationOnce(() => product);
-    const result = await main(event, this, () => { }) as APIGatewayProxyResult;
+    jest.spyOn(productsActions, 'getProductsById').mockImplementationOnce(() => Promise.resolve(product));
```

Comment:
nit: you can use mockResolvedValue or mockResolvedValueOnce as a shorthand for mockImplementationOnce(() => Promise.resolve(product));

https://jestjs.io/docs/mock-function-api#mockfnmockresolvedvalueoncevalue